cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueAdmin.java

```diff
@@ -137,15 +138,21 @@ protected final synchronized HBaseAdmin getHBaseAdmin() throws IOException {
    */
   public String getActualTableName(QueueName queueName) {
     if (queueName.isQueue()) {
-      // <cdap namespace>.system.queue.<account>.<flow>
-      return getTableNameForFlow(queueName.getFirstComponent(), queueName.getSecondComponent());
+      // <cdap namespace>.system.queue.<namespace>.<app>.<flow>
+      return getTableNameForFlow(queueName.getFirstComponent(),
+                                 queueName.getSecondComponent(),
+                                 queueName.getThirdComponent());
     } else {
       throw new IllegalArgumentException("'" + queueName + "' is not a valid name for a queue.");
     }
   }
 
-  private String getTableNameForFlow(String app, String flow) {
-    return tableNamePrefix + "." + app + "." + flow;
+  private String getTableNameForFlow(String namespaceId, String app, String flow) {
+    StringBuilder tableName = new StringBuilder(tableNamePrefix).append(".");
+    if (!Constants.DEFAULT_NAMESPACE.equals(namespaceId)) {
```

Comment:
We do not want to add ``[namespace-id]`` to queues in ``default`` namespace so as to make upgrade easier. 

In 2.6.0, we do not have namespaceId/accountId in these table names. We want to support existing queues in 2.8.0 as well.